akka-stream/src/main/scala/akka/stream/impl/ActorRefSinkStage.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.impl
+
+import akka.actor.{ ActorRef, Terminated }
+import akka.annotation.InternalApi
+import akka.stream.impl.Stages.DefaultAttributes
+import akka.stream.{ AbruptStageTerminationException, Attributes, Inlet, SinkShape }
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, StageLogging }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+final private[akka] class ActorRefSinkStage[T](
+    ref: ActorRef,
+    onCompleteMessage: Any,
+    onFailureMessage: Throwable => Any)
+    extends GraphStage[SinkShape[T]] {
+
+  val in: Inlet[T] = Inlet("ActorRefSink.in")
+
+  override def shape: SinkShape[T] = SinkShape(in)
+
+  override protected def initialAttributes: Attributes = DefaultAttributes.actorRefSink
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with InHandler with StageLogging {
+
+      var completionSignalled = false
+
+      override def preStart(): Unit = {
+        getStageActor({
+          case (_, Terminated(`ref`)) =>
+            completeStage()
+          case msg =>
+            log.error("Unexpected message to stage actor {}", msg)
```

Comment:
we try to avoid `toString` logging of unknown messages, the message class would be enough